examples/angular2/services/store.ts

```diff
@@ -0,0 +1,87 @@
+//http://stackoverflow.com/questions/105035/create-guid-uuid-in-javascript
+
+function generateUID () {
+	function s4 () {
+		return Math.floor((1 + Math.random()) * 0x10000)
+			.toString(16)
+			.substring(1);
+	}
+	return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
+		s4() + '-' + s4() + s4() + s4();
+}
+
+export class Todo {
+	completed: Boolean;
+	editing: Boolean;
+	text: String;
+	uid: String;
+	cleanText(text:String) {
+		return text.replace(/\W+$/, '');
+	}
+	setText(text:String) {
+		this.text = this.cleanText(text);
+
+		if (this.text.length == 0) {
+			return false;
+		}
+	}
+	constructor(text:String) {
+		this.uid = generateUID();
+		this.completed = false;
+		this.editing = false;
+		this.text = this.cleanText(text);
+	}
+}
+
+export class TodoStore {
+	todos: Array<Todo>;
+	constructor() {
+		this.todos = (store.get('angular2-todos') || []);
```

Comment:
I wonder if the wrapping in parenthesises makes much sense. Doesn't really improve readability does it?